vscode/src/editor/active-editor.ts

```diff
@@ -0,0 +1,36 @@
+import * as vscode from 'vscode'
+
+/**
+ * Gets the currently active text editor instance if available.
+ * Returns undefined if no editor is active.
+ *
+ * NOTE: This handles edge case where activeTextEditor API returns undefined when webview panel has focus.
+ */
+let lastTrackedTextEditor: vscode.TextEditor | undefined
+
+export function getActiveEditor(): vscode.TextEditor | undefined {
+    // When there is no active editor, reset lastTrackedTextEditor
+    const activeEditors = vscode.window.visibleTextEditors
+    if (!activeEditors.length) {
+        lastTrackedTextEditor = undefined
+        return undefined
+    }
+
+    // When webview panel is focus, calling activeTextEditor will return undefined.
+    // This allows us to get the active editor before the webview panel is focused.
+    const get = (): vscode.TextEditor | undefined => {
+        const activeEditor = vscode.window.activeTextEditor
+        if (activeEditor?.document.uri.scheme === 'file') {
+            lastTrackedTextEditor = activeEditor
+        }
+        return activeEditor || lastTrackedTextEditor
+    }
+
+    return get()
```

Comment:
Will it work the same way?

```suggestion
    const activeEditor = vscode.window.activeTextEditor
    if (activeEditor?.document.uri.scheme === 'file') {
        lastTrackedTextEditor = activeEditor
    }
    
    return activeEditor || lastTrackedTextEditor
```